src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalNeeded.tsx

```diff
@@ -11,44 +12,49 @@ import { GetAllowanceErr } from './helpers'
 export const useIsApprovalNeeded = (
   tradeQuoteStep: TradeQuoteStep | undefined,
   sellAssetAccountId: AccountId | undefined,
-  watch: boolean,
 ) => {
-  const { isLoading, data } = useAllowance(tradeQuoteStep, sellAssetAccountId, watch)
-
-  const isApprovalNeeded = useMemo(() => {
-    if (tradeQuoteStep === undefined) return undefined
-
-    const isEvmChainId = evmChainIds.includes(tradeQuoteStep.sellAsset.chainId as EvmChainId)
-
-    if (!isEvmChainId) return false
-
-    if (data?.isErr()) {
-      const error = data.unwrapErr()
-      // the error type is a GetAllowanceErr enum so we can handle all cases with exhaustiveness
-      // checking to prevent returning the wrong value if we add more error cases
-      switch (error) {
-        case GetAllowanceErr.IsFeeAsset:
-        case GetAllowanceErr.NotEVMChain:
-          // approval not required
-          return false
-        case GetAllowanceErr.MissingArgs:
-          // not known yet
-          return undefined
-        default:
-          assertUnreachable(error)
+  const selectIsApprovalNeeded = useCallback(
+    (data: Result<string, GetAllowanceErr>) => {
+      if (tradeQuoteStep === undefined) return undefined
+
+      if (data?.isErr()) {
+        const error = data.unwrapErr()
+        // the error type is a GetAllowanceErr enum so we can handle all cases with exhaustiveness
+        // checking to prevent returning the wrong value if we add more error cases
+        switch (error) {
+          case GetAllowanceErr.IsFeeAsset:
+          case GetAllowanceErr.NotEVMChain:
+            // approval not required
+            return false
+          default:
+            assertUnreachable(error)
+        }
       }
-    }
 
-    const allowanceOnChainCryptoBaseUnit = data?.unwrap()
-    return allowanceOnChainCryptoBaseUnit !== undefined
-      ? bn(allowanceOnChainCryptoBaseUnit).lt(
-          tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
-        )
-      : undefined
-  }, [data, tradeQuoteStep])
+      const allowanceOnChainCryptoBaseUnit = data?.unwrap()
+      return allowanceOnChainCryptoBaseUnit !== undefined
+        ? bn(allowanceOnChainCryptoBaseUnit).lt(
+            tradeQuoteStep.sellAmountIncludingProtocolFeesCryptoBaseUnit,
+          )
+        : undefined
+    },
+    [tradeQuoteStep],
+  )
+
+  const { data: isApprovalNeeded, isLoading: isApprovalNeededLoading } = useQuery({
+    ...reactQueries.common.allowanceCryptoBaseUnit(
+      tradeQuoteStep?.sellAsset.assetId,
+      tradeQuoteStep?.allowanceContract,
+      sellAssetAccountId ? fromAccountId(sellAssetAccountId)?.account : undefined,
+    ),
+    refetchInterval: 15_000,
```

Comment:
@woodenfurniture to shine here - we may be able to do one even better by introspecting the approval txId - if it is, and it is confirmed, we could `invalidateQueries` instead of being refetchy, making things even faster for fast chains - same like LP is doing, which is why it had a higher 30s `refetchInterval`, although now that `useAllowance` is abstracted away, the interval is non-configurable.
We could possibly pass query options in to make it configurable per query, but a better option would be to keep it at 30s for all consumers, and only `invalidateQueries` on success Tx here